pybamm/install_odes.py

```diff
@@ -127,7 +127,7 @@ def main(arguments=None):
     desc = "Install scikits.odes."
     parser = argparse.ArgumentParser(description=desc)
     parser.add_argument("--sundials-libs", type=str, help="path to sundials libraries.")
-    default_install_dir = os.path.join(os.getenv("HOME"), ".local")
+    default_install_dir = os.path.join("/home/runner", ".local")
```

Comment:
Hardcoding the env variable will fail the script locally. Is this script passing locally?

My idea was to pass `HOME` in the `testenv` environment specified in `tox.ini`. Something like this - 

https://github.com/pybamm-team/PyBaMM/blob/57d06db6a6da0cf5a304639df29163dd8e2113fb/tox.ini#L36

Also, the tests are passing because the `testenv` environment is reading SUNDIALS and scikits.odes from `pybamm-requires`, as the logs say -
```
odes run-test-pre: PYTHONHASHSEED='645949218'
odes run-test: commands[0] | sh -c pybamm_install_odes
Requirement already satisfied: scikits.odes in ./.tox/odes/lib/python3.10/site-packages (2.7.0)
Requirement already satisfied: scipy in ./.tox/odes/lib/python3.10/site-packages (from scikits.odes) (1.10.1)
Requirement already satisfied: numpy<1.27.0,>=1.19.5 in ./.tox/odes/lib/python3.10/site-packages (from scipy->scikits.odes) (1.24.3)
/home/runner/work/PyBaMM/PyBaMM/pybamm/scikits_odes_setup.log
/home/runner/.local/lib was found in LD_LIBRARY_PATH.
--> Not updating venv activate or .bashrc scripts
```
Can you maybe try removing them in the workflows or the tox environment before `pybamm_install_odes` is tested?

Reply:
My bad, I'm now passing `HOME` in the `passenv`. \
\
But as I speak about  SUNDIALS & `scikit.odes` being in `pybamm-requires`,
I cannot find these two locally inside `pybamm-requires` so that i could try removing them before testing, but in `odes` (which is being created while testing pybamm_install_odes only) I can find `scikit.odes` and it's other dependencies. 
\
![image](https://github.com/pybamm-team/PyBaMM/assets/104268427/8f03645b-8039-4c91-83cb-4376c1b2e2a3)

Reply:
You can simply do `pip uninstall --yes scikits.odes` to remove the python library. `SUNDIALS` is not a python library hence it will not be present in the `site-packages` directory; the path should be present in `scikits_odes_setup.log`.

Reply:
> You can simply do `pip uninstall --yes scikits.odes` to remove the python library. `SUNDIALS` is not a python library hence it will not be present in the `site-packages` directory; the path should be present in `scikits_odes_setup.log`.

I have handled this, all tests are passing locally. I am now running scheduled tests to make sure things are right.

Reply:
Here are the [tests](https://github.com/arjxn-py/PyBaMM/actions/runs/5109964949) @Saransh-cpp